### PR TITLE
[core] Catalog support list `sys` namespace for end user

### PR DESCRIPTION
**Pull request by ulysses-you**
### Purpose

This pr adds `sys` database when calling `listDatabases(includeSystem)` in `Catalog`, so that user can be aware of the Paimon sys database featrue.

However, it seems `listDatabases` is used by Paimon internal, e.g., list all table path, so only support catalog `listNamespaces` to only affects end-user.

### Tests

add test

### API and Format

no

### CODE REVIEW

**Comment by ulysses-you** [comment]
cc @JingsongLi thank you

File: paimon-core/src/main/java/org/apache/paimon/catalog/Catalog.java [hunk 1]
```diff
@@ -84,10 +85,23 @@ default Optional<MetastoreClient.Factory> metastoreClientFactory(Identifier iden
     /**
      * Get the names of all databases in this catalog.
      *
-     * @return a list of the names of all databases
+     * @return a list of the names of user databases
      */
     List<String> listDatabases();
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/catalog/Catalog.java:90]: It seems we should modify the implementation of `listDatabases`?

Hi @ulysses-you 
Could you research whether system database in database systems can be viewed with a list database command, such as in MySQL, PostgreSQL, etc.

Comment (ulysses-you) [reply paimon-core/src/main/java/org/apache/paimon/catalog/Catalog.java:90]: yea, I tried mysql and PostgreSQL, both of them will show the sys databases if user has permission.
```sql
mysql> show databases;
+--------------------+
| Database           |
+--------------------+
| information_schema |
| mysql              |
| performance_schema |
| sys                |
+--------------------+
```

```sql
postgres=# \l
                                            数据库列表
   名称    | 拥有者 | 字元编码 | 校对规则 | Ctype | ICU Locale | Locale Provider |    存取权限
-----------+--------+----------+----------+-------+------------+-----------------+-----------------
 postgres  | cathy  | UTF8     | C        | C     |            | libc            |
 template0 | cathy  | UTF8     | C        | C     |            | libc            | =c/cathy       +
           |        |          |          |       |            |                 | cathy=CTc/cathy
 template1 | cathy  | UTF8     | C        | C     |            | libc            | =c/cathy       +
           |        |          |          |       |            |                 | cathy=CTc/cathy
(3 行记录)

```

Comment (JingsongLi) [reply paimon-core/src/main/java/org/apache/paimon/catalog/Catalog.java:90]: Thanks @ulysses-you 

We should implement `listDatabases` in `AbstractCatalog`.

Comment (ulysses-you) [reply paimon-core/src/main/java/org/apache/paimon/catalog/Catalog.java:90]: @JingsongLi  How about change ` List<String> listDatabases();` to ` List<String> listDatabases(boolean includeSystem);`? I'm afraid always return sys database will break somethings. I beleive the usage of `listDatabases` Paimon internal do not need sys database.

**Comment by JingsongLi** [comment]
Hi @ulysses-you , where is this requirement from? If this requirement doesn't come from the user, I suggest we don't return the sys db. I checked that Iceberg also doesn't return it.

**Comment by ulysses-you** [comment]
I see, leave it as is for now. thank you @JingsongLi

**Pull request closed without merging**